.github/workflows/release.yml

```diff
@@ -6,46 +6,103 @@ on:
       version:
         description: 'Go version to build ("1.17")'
         required: true
+      prerelease:
+        description: 'Is this a pre-release?'
+        required: true
+        default: 'false'
 
 jobs:
   build:
     strategy:
       matrix:
         include:
-          - builder: ubuntu-latest
-            goos: linux
-            goarch: amd64
           - builder: macos-latest
             goos: darwin
-            goarch: amd64
+            goarch: arm64
           - builder: macos-latest
             goos: darwin
-            goarch: arm64
+            goarch: amd64
+          - builder: ubuntu-latest
+            goos: linux
+            goarch: amd64
           - builder: windows-latest
             goos: windows
             goarch: amd64
 
     runs-on: ${{ matrix.builder }}
+    outputs:
+      built_version: ${{ steps.encore_go_version.outputs.version }}
     steps:
       - name: Check out repo
         uses: actions/checkout@v2
+        with:
+          fetch-depth: 0   # We need the full history for the Go submodule
+          submodules: true # Checkout the Go submodule
 
       - name: Set up Go
         uses: actions/setup-go@v2
         with:
           go-version: 1.17
+
       - name: 'Create patched runtime'
         run: bash ./apply_patch.bash "${{ github.event.inputs.version }}"
-        env:
-          GO111MODULE: "on"
+
       - name: Build
         run: go run . -dst=dist -goos=${{ matrix.goos }} -goarch=${{ matrix.goarch }}
         env:
           GO111MODULE: "on"
+
+      # This step gets the exact version of Go we're releasing (including minor), so if we give the input as `1.17` this might return `encore-go1.17.5`.
+      - name: 'Read the version of Go we built'
+        id: encore_go_version
+        run: echo "::set-output name=version::$(go run . --read-built-version)"
```

Comment:
Note; I did this via the `main.go` file so it works on Windows as well as MacOS & Linux (and doesn't need us to write separate powershell code vs bash code)